Installation issue: kealib (not recognizing C compiler)

### Steps to reproduce the issue

```console
$ spack --env zen2-gcc13 find -lvfdIc  kealib
...
-- linux-rhel8-zen2 / %cxx=gcc@13.4.0 ---------------------------
 -   q2nlf2w kealib@1.5.3+ipo build_system=cmake build_type=RelWithDebInfo generator=make
[+]  eufn52p     cmake@3.31.9~doc+ncurses+ownlibs~qtgui build_system=generic build_type=RelWithDebInfo
[+]  gu5vdnc         curl@8.15.0+gssapi~ldap+libidn2~librtmp~libssh+libssh2+nghttp2 build_system=autotools libs:=shared,static tls:=mbedtls,openssl
[+]  r2dc5ck             krb5@1.21.3+shared build_system=autotools
[+]  kdrpmhr                 bison@3.8.2~color build_system=autotools
[+]  vlbkf4y                     m4@1.4.20+sigsegv build_system=autotools
[+]  focymqv                         libsigsegv@2.14 build_system=autotools
[+]  iartorz                 diffutils@3.12 build_system=autotools
[+]  zfcjhbz                     libiconv@1.18 build_system=autotools libs:=shared,static
[+]  zirylds                 findutils@4.10.0 build_system=autotools patches:=440b954
[+]  ful6i3c                 gettext@0.23.1+bzip2+curses+git+libunistring+libxml2+pic+shared+tar+xz build_system=autotools
[+]  btav5ox                     libxml2@2.13.5+http+pic~python+shared build_system=autotools
[+]  ahyoscx                     tar@1.35 build_system=autotools zip=pigz
[+]  vxpii7z                         pigz@2.8 build_system=makefile
[+]  n64duty                         zstd@1.5.7+programs build_system=makefile compression:=lz4,lzma,zlib libs:=shared,static
[+]  3367tmt                             lz4@1.10.0+pic build_system=makefile libs:=shared,static
[+]  ihgv3uq                 libedit@3.1-20240808 build_system=autotools
[+]  xtgeq4a             libidn2@2.3.7 build_system=autotools
[+]  xbdx6lg                 libunistring@1.2 build_system=autotools
[+]  24bzzhf             libssh2@1.11.1+shared build_system=autotools crypto=mbedtls
[+]  pyx73rp                 xz@5.6.3+pic build_system=autotools libs:=shared,static
[+]  crudhjn             mbedtls@2.28.9+pic build_system=makefile build_type=RelWithDebInfo libs:=shared,static
[+]  7wi62dj             nghttp2@1.67.1 build_system=autotools
[+]  unrffxy             openssl@3.6.0~docs+shared build_system=generic certs=system
[+]  edlxjs4             perl@5.42.0+cpanm+opcode+open+shared+threads build_system=generic
[+]  pxa3pju                 berkeley-db@18.1.40+cxx+docs+stl build_system=autotools patches:=26090f4
[+]  jiyqkwo                 bzip2@1.0.8~debug+pic+shared build_system=generic
[+]  m4ifnmh                 gdbm@1.25 build_system=autotools
[+]  f375v3a                     readline@8.3 build_system=autotools patches:=21f0a03
[+]  vet42cx         ncurses@6.5-20250705+symlinks+termlib abi=5 build_system=autotools patches:=7a351bc
[+]  p4i3vud         zlib@1.3.1+optimize+pic+shared build_system=makefile
[+]  c4lvqa5     compiler-wrapper@1.0 build_system=generic
[e]  gw3zygo     gcc@13.4.0+binutils~bootstrap+graphite~mold+nvptx~piclibs~strip build_system=autotools build_type=RelWithDebInfo languages:='c,c++,fortran,lto'
[+]  gd56l37     gcc-runtime@13.4.0 build_system=generic
[e]  3kzdcra     glibc@2.28 build_system=autotools
[+]  ymb7dyf     gmake@4.4.1~guile build_system=generic
[+]  qgkirix     hdf5@1.14.6+cxx+fortran+hl+ipo~java+map~mpi+shared+szip+threadsafe+tools api=default build_system=cmake build_type=RelWithDebInfo cxxstd=11 generator=make
[+]  gegxhra         libaec@1.1.4+ipo+shared build_system=cmake build_type=RelWithDebInfo generator=make
[+]  2pp7u3i         pkgconf@2.5.1 build_system=autotools
```
(full output is at [spec.kealib.txt](https://github.com/user-attachments/files/28362950/spec.kealib.txt))

### Error message

<details><summary>Error message</summary>

<pre>
==> No patches needed for kealib
==> kealib: Executing phase: 'cmake'
==> [2026-05-27-15:15:21.610636] /cvmfs/hpcsw.umd.edu/spack-software/2025.11.0/linux-zen2/cmake-3.31.9-eufn52pskigbs2uuke5vjk65qakls5vk/bin/cmake -G 'Unix Makefiles' -DCMAKE_INSTALL_PREFIX:STRING=/cvmfs/hpcsw.umd.edu/spack-software/2025.11.0/linux-zen2/kealib-1.5.3-q2nlf2wmvhpfoavmd2cz6sawbjbu72vq -DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=ON '-DCMAKE_INSTALL_RPATH:STRING=/cvmfs/hpcsw.umd.edu/spack-software/2025.11.0/linux-zen2/kealib-1.5.3-q2nlf2wmvhpfoavmd2cz6sawbjbu72vq/lib;/cvmfs/hpcsw.umd.edu/spack-software/2025.11.0/linux-zen2/kealib-1.5.3-q2nlf2wmvhpfoavmd2cz6sawbjbu72vq/lib64' '-DCMAKE_PREFIX_PATH:STRING=/cvmfs/hpcsw.umd.edu/spack-software/2025.11.0/linux-zen2/cmake-3.31.9-eufn52pskigbs2uuke5vjk65qakls5vk;/cvmfs/hpcsw.umd.edu/spack-software/2025.11.0/linux-zen2/compiler-wrapper-1.0-c4lvqa5ekdyxfibdcfh6hfepzpfknhp5;/cvmfs/hpcsw.umd.edu/spack-software/2025.11.0/linux-zen2/gmake-4.4.1-ymb7dyfcziesq5brhndx4pxrqb2g4l25;/cvmfs/hpcsw.umd.edu/spack-software/2025.11.0/linux-zen2/hdf5-1.14.6-qgkirixzo4gkny5fcmastcs2usi2vt4m;/cvmfs/hpcsw.umd.edu/spack-software/2025.11.0/linux-zen2/libaec-1.1.4-gegxhraj7xpyg5zhi4hcjnq73m573g3b;/cvmfs/hpcsw.umd.edu/spack-software/2025.11.0/linux-zen2/zlib-1.3.1-p4i3vudzzs4cjl6bkziqjpxi7arstkc7;/cvmfs/hpcsw.umd.edu/spack-software/2025.11.0/linux-zen2/gcc-runtime-13.4.0-gd56l37g5fvduuy224vwdotkksk2c2lk;/cvmfs/hpcsw.umd.edu/gcc/13.4.0/rhel8-x86_64' -DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON -DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=ON -DCMAKE_POLICY_DEFAULT_CMP0090:STRING=NEW -DCMAKE_FIND_USE_PACKAGE_REGISTRY:BOOL=OFF -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=ON -DHDF5_ROOT=/cvmfs/hpcsw.umd.edu/spack-software/2025.11.0/linux-zen2/hdf5-1.14.6-qgkirixzo4gkny5fcmastcs2usi2vt4m /tmp/swinst26/spackswinst26/spack-stage/spack-stage-kealib-1.5.3-q2nlf2wmvhpfoavmd2cz6sawbjbu72vq/spack-src
...
-- The C compiler identification is unknown
-- The CXX compiler identification is GNU 13.4.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - failed
-- Check for working C compiler: /cvmfs/hpcsw.umd.edu/spack-software/2025.11.0/linux-zen2/compiler-wrapper-1.0-c4lvqa5ekdyxfibdcfh6hfepzpfknhp5/libexec/spack/cc
-- Check for working C compiler: /cvmfs/hpcsw.umd.edu/spack-software/2025.11.0/linux-zen2/compiler-wrapper-1.0-c4lvqa5ekdyxfibdcfh6hfepzpfknhp5/libexec/spack/cc - broken
CMake Error at /cvmfs/hpcsw.umd.edu/spack-software/2025.11.0/linux-zen2/cmake-3.31.9-eufn52pskigbs2uuke5vjk65qakls5vk/share/cmake-3.31/Modules/CMakeTestCCompiler.cmake:67 (message):
  The C compiler

    "/cvmfs/hpcsw.umd.edu/spack-software/2025.11.0/linux-zen2/compiler-wrapper-1.0-c4lvqa5ekdyxfibdcfh6hfepzpfknhp5/libexec/spack/cc"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: '/tmp/swinst26/spackswinst26/spack-stage/spack-stage-kealib-1.5.3-q2nlf2wmvhpfoavmd2cz6sawbjbu72vq/spack-build-q2nlf2w/CMakeFiles/CMakeScratch/TryCompile-ZzoRZx'

    Run Build Command(s): /cvmfs/hpcsw.umd.edu/spack-software/2025.11.0/linux-zen2/cmake-3.31.9-eufn52pskigbs2uuke5vjk65qakls5vk/bin/cmake -E env VERBOSE=1 /cvmfs/hpcsw.umd.edu/spack-software/2025.11.0/linux-zen2/gmake-4.4.1-ymb7dyfcziesq5brhndx4pxrqb2g4l25/bin/gmake -f Makefile cmTC_f2b48/fast
    /cvmfs/hpcsw.umd.edu/spack-software/2025.11.0/linux-zen2/gmake-4.4.1-ymb7dyfcziesq5brhndx4pxrqb2g4l25/bin/gmake  -f CMakeFiles/cmTC_f2b48.dir/build.make CMakeFiles/cmTC_f2b48.dir/build
    gmake[1]: Entering directory '/tmp/swinst26/spackswinst26/spack-stage/spack-stage-kealib-1.5.3-q2nlf2wmvhpfoavmd2cz6sawbjbu72vq/spack-build-q2nlf2w/CMakeFiles/CMakeScratch/TryCompile-ZzoRZx'
    Building C object CMakeFiles/cmTC_f2b48.dir/testCCompiler.c.o
    /cvmfs/hpcsw.umd.edu/spack-software/2025.11.0/linux-zen2/compiler-wrapper-1.0-c4lvqa5ekdyxfibdcfh6hfepzpfknhp5/libexec/spack/cc    -o CMakeFiles/cmTC_f2b48.dir/testCCompiler.c.o -c /tmp/swinst26/spackswinst26/spack-stage/spack-stage-kealib-1.5.3-q2nlf2wmvhpfoavmd2cz6sawbjbu72vq/spack-build-q2nlf2w/CMakeFiles/CMakeScratch/TryCompile-ZzoRZx/testCCompiler.c
    [spack cc]: Error: SPACK_CC_* variables not set: this usually means a missing `depends_on("c", type="build")` in package.py
    gmake[1]: *** [CMakeFiles/cmTC_f2b48.dir/build.make:81: CMakeFiles/cmTC_f2b48.dir/testCCompiler.c.o] Error 1
    gmake[1]: Leaving directory '/tmp/swinst26/spackswinst26/spack-stage/spack-stage-kealib-1.5.3-q2nlf2wmvhpfoavmd2cz6sawbjbu72vq/spack-build-q2nlf2w/CMakeFiles/CMakeScratch/TryCompile-ZzoRZx'
    gmake: *** [Makefile:134: cmTC_f2b48/fast] Error 2
</pre></details>


### Information on your system

```console
* **Spack:** 1.2.0.dev0
* **Builtin repo:** https://github.com/spack/spack-packages/commit/119680aeee8ea802c6111b7167583bddef97e82f
* **Python:** 3.6.8
* **Platform:** linux-rhel8-zen2
```

### Additional information

- [spack-build-out.txt](https://github.com/user-attachments/files/28362773/spack-build-out.txt)
- [spack-build-env.txt](https://github.com/user-attachments/files/28362779/spack-build-env.txt)
- [spack-build-env-mods.txt](https://github.com/user-attachments/files/28362784/spack-build-env-mods.txt)

Maintainers:
- @gillins  
- @neilflood  
- @petebunting

The kealib package.py only gives cxx as a dependency, it looks to me like it depends on c as well (or at least some of the Cmake infrastructure it uses depends on c).

### General information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [x] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

--- a/repos/spack_repo/builtin/packages/kealib/package.py
+++ b/repos/spack_repo/builtin/packages/kealib/package.py
@@ -45,8 +45,8 @@ class Kealib(CMakePackage):
     version("1.4.8", sha256="0f24d8478865abcb17865c8f49c0370095726c529b8ac373ffae018ad3d40a02")
     version("1.4.7", sha256="ec38751b3b555d3a26f0c7445f2d2cd9d7c3a3502237519a206a50cb58df56ec")
 
-    depends_on("cxx", type="build")  # generated
-    depends_on("c", type="build")  # Issue #5006
+    depends_on("c", type="build")
+    depends_on("cxx", type="build")
 
     depends_on("cmake@3.5:", type="build")
     depends_on("hdf5+cxx+hl", when="@:1.5.1")

--- a/repos/spack_repo/builtin/packages/kealib/package.py
+++ b/repos/spack_repo/builtin/packages/kealib/package.py
@@ -46,6 +46,7 @@ class Kealib(CMakePackage):
     version("1.4.7", sha256="ec38751b3b555d3a26f0c7445f2d2cd9d7c3a3502237519a206a50cb58df56ec")
 
     depends_on("cxx", type="build")  # generated
+    depends_on("c", type="build")  # Issue #5006
 
     depends_on("cmake@3.5:", type="build")
     depends_on("hdf5+cxx+hl", when="@:1.5.1")

